errors/registry.go

```diff
@@ -0,0 +1,109 @@
+// Copyright © 2017 The Things Network
+// Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+
+package errors
+
+import (
+	"fmt"
+	"sync"
+)
+
+// registry represents an error type registry
+type registry struct {
+	sync.RWMutex
+	byCode map[Code]*ErrDescriptor
+}
+
+// Register registers a new error type
+func (r *registry) Register(err *ErrDescriptor) {
+	r.Lock()
+	defer r.Unlock()
+
+	if err.Code == 0 {
+		panic(fmt.Errorf("No code defined in error descriptor (message: `%s`)", err.MessageFormat))
+	}
+
+	if r.byCode[err.Code] != nil {
+		panic(fmt.Errorf("errors: Duplicate error code %v registered", err.Code))
+	}
+
+	err.registered = true
+	r.byCode[err.Code] = err
+}
+
+// Get returns the descriptor if it exists or nil otherwise
+func (r *registry) Get(code Code) *ErrDescriptor {
+	r.RLock()
+	defer r.RUnlock()
+	return r.byCode[code]
+}
+
+// reg is a global registry to be shared by packages
```

Comment:
Even though I am not sure how sane it sounds, we could add an endpoint for components, which, given the error code returned by the component, would return the `ErrDescriptor`. This could be stored in an internal hashmap of the client and once a new error code is encountered, it would look it up again.

* Console requests foo from NOC, gets error with code 42
* Console looks up internal hashmap of ErrDescriptors for NOC, code 42 not found
* Console executes `ErrDescriptor(42)` RPC to NOC and gets the error descriptor, stores that in internal hashmap

Given backwards stability of API that would work. This procedure could be handled by this package.

We might also add some "well-known codes", for which lookups are not required